lib/postgrex/extensions/tsvector.ex

```diff
@@ -0,0 +1,96 @@
+defmodule Postgrex.Extensions.Tsvector do
+  @moduledoc false
+
+  import Postgrex.BinaryUtils, warn: false
+  use Bitwise
+  use Postgrex.BinaryExtension, [send: "tsvectorsend"]
+
+  def encode(_) do
+    quote location: :keep do
+      values when is_list(values) ->
+        encoded_tsvectors = unquote(__MODULE__).encode_tsvector(values)
+        <<byte_size(encoded_tsvectors)::int32, encoded_tsvectors::binary>>
+      other ->
+        raise ArgumentError,
+        Postgrex.Utils.encode_msg(other, "a list of tsvectors")
+    end
+  end
+
+  def decode(_) do
+    quote do
+      <<len::int32, value::binary-size(len)>> ->
+        <<nb_lexemes::int32, words::binary>> = value
+        unquote(__MODULE__).decode_tsvector_values(:binary.bin_to_list(words))
+    end
+  end
+
+  ## Helpers
+
+  def encode_tsvector(values) do
+    <<length(values)::int32, encode_lexemes(values)::binary>>
+  end
+
+  def encode_lexemes(values) do
+    Enum.map(values, fn(x) -> encode_positions(x) end) |> IO.iodata_to_binary
+  end
+
+  defp encode_positions(%Postgrex.Lexeme{word: word, positions: positions}) do
+    positions = Enum.map(positions, fn({position, weight}) -> <<encode_weight_binary(weight)::2, position::14>> end)
+    [[word, 0], <<length(positions)::16>>, positions]
+  end
+
+  def decode_tsvector_values(values) do decode_tsvector_values(values, []) end
+
+  defp decode_tsvector_values(data, acc) do
+    decode_tsvector_values data, acc, Enum.find_index(data, fn(x) -> x == 0 end)
+  end
+
+  defp decode_tsvector_values(_, acc, word_end_index) when word_end_index == nil, do: acc
+
+  defp decode_tsvector_values(data, acc, word_end_index) do
+    decode_tsvector_values(data, acc, word_end_index, positions_end_index(data, word_end_index))
+  end
+
+  defp decode_tsvector_values(data, acc, word_end_index, lexeme_end_index) when lexeme_end_index == 0 do
+    word = Enum.take(data, word_end_index) |> IO.iodata_to_binary
+    lexeme = %Postgrex.Lexeme{word: word, positions: []}
+    decode_tsvector_values(Enum.drop(data, word_end_index + 3), acc ++ [lexeme])
+  end
+
+  defp decode_tsvector_values(data, acc, word_end_index, lexeme_end_index) do
+    word = Enum.take(data, word_end_index) |> IO.iodata_to_binary
+    positions = extract_positions(Enum.slice(data, word_end_index + 3, (lexeme_end_index - (word_end_index + 3))))
+    lexeme = %Postgrex.Lexeme{word: word, positions: positions}
+    decode_tsvector_values(Enum.drop(data, lexeme_end_index), acc ++ [lexeme])
+  end
+
+  defp extract_positions(bytes) do
+    Enum.chunk(bytes, 2)
+    |> Enum.map(fn(x) ->
+        <<raw_word_meta::uint16>> = :erlang.list_to_bitstring(x)
+        {raw_word_meta &&& 0x3fff, decode_weight(raw_word_meta >>> 14)}
+       end)
+  end
+
+  defp encode_weight_binary('A') do 3 end
```

Comment:
Well :A might be faster because comparison of atoms is as quick as it gets but it string or atoms felt more appropriate from language perspective. Atom because there are only 3 named weights. String because its used more than char list. Lets see what another maintainer thinks.